packages/flutter/lib/src/widgets/app.dart

```diff
@@ -134,25 +182,103 @@ class WidgetsApp extends StatefulWidget {
   /// During normal app operation, the [onGenerateRoute] callback will only be
   /// applied to route names pushed by the application, and so should never
   /// return null.
+  ///
+  /// This is used if [routes] does not contain the requested route.
+  ///
+  /// The [Navigator] is only built if routes are provided (either via [home],
+  /// [routes], [onGenerateRoute], or [onUnknownRoute]); if they are not,
+  /// [builder] must not be null.
   /// {@endtemplate}
   ///
-  /// The [Navigator] is only built if [onGenerateRoute] is not null. If
-  /// [onGenerateRoute] is null, the [builder] must be non-null.
+  /// If this property is not set, either the [routes] or [home] properties must
+  /// be set, and the [pageRouteBuilder] must also be set so that the
+  /// default handler will know what routes and [PageRoute]s to build.
   final RouteFactory onGenerateRoute;
 
-  /// Called when [onGenerateRoute] fails to generate a route.
+  /// The [PageRoute] generator callback used when the app is navigated to a
+  /// named route.
+  ///
+  /// This callback can be used, for example, to specify that a [MaterialPageRoute]
+  /// or a [CupertinoPageRoute] should be used for building page transitions.
+  final PageRouteFactory pageRouteBuilder;
+
+  /// {@template flutter.widgets.widgetsApp.home}
+  /// The widget for the default route of the app ([Navigator.defaultRouteName],
+  /// which is `/`).
   ///
+  /// This is the route that is displayed first when the application is started
+  /// normally, unless [initialRoute] is specified. It's also the route that's
+  /// displayed if the [initialRoute] can't be displayed.
+  ///
+  /// To be able to directly call [Theme.of], [MediaQuery.of], etc, in the code
+  /// that sets the [home] argument in the constructor, you can use a [Builder]
+  /// widget to get a [BuildContext].
+  ///
+  /// If [home] is specified, then [routes] must not include an entry for `/`,
+  /// as [home] takes its place.
+  ///
+  /// The [Navigator] is only built if routes are provided (either via [home],
+  /// [routes], [onGenerateRoute], or [onUnknownRoute]); if they are not,
+  /// [builder] must not be null.
+  ///
+  /// The difference between using [home] and using [builder] is that the [home]
+  /// subtree is inserted into the application below a [Navigator] (and thus
+  /// below an [Overlay], which [Navigator] uses). With [home], therefore,
+  /// dialog boxes will work automatically, the [routes] table will be used, and
+  /// APIs such as [Navigator.push] and [Navigator.pop] will work as expected.
+  /// In contrast, the widget returned from [builder] is inserted _above_ the
```

Comment:
`inserted _above_ the app's [Navigator]`? To compromise between the specific docs and a generalized doc.

Reply:
Done

Reply:
Done